File: trees_and_graphs/binary_tree_right_side_view.java
public List<Integer> rightSideView(TreeNode root)
{
    if (root == null)
    {
        return new ArrayList<>();
    }

    List<Integer> ans = new ArrayList<>();
    Queue<TreeNode> queue = new LinkedList<>();
    queue.add(root);

    while (!queue.isEmpty())
    {
        int currentLength = queue.size();
        int prev = 0;

        for (int i = 0; i < currentLength; i++)
        {
            TreeNode node = queue.remove();
            prev = node.val;

            if (node.left != null)
            {
                queue.add(node.left);
            }

            if (node.right != null)
            {
                queue.add(node.right);
            }
        }

        ans.add(prev);
    }

    return ans;
}

File: trees_and_graphs/binary_tree_right_side_view.py
import treenode
from collections import deque

def rightSideView(root: treenode.TreeNode) -> List[int]:
    if not root:
        return []
    
    ans = []
    queue = deque([root])

    while queue:
        current_length = len(queue)
        ans.append(queue[-1].val)

        for _ in range(current_length):
            node = queue.popleft()

            if node.left:
                queue.append(node.left)

            if node.right:
                queue.append(node.right)
    
    return ans
